core/src/main/java/org/apache/accumulo/core/util/ratelimit/SharedRateLimiterFactory.java

```diff
@@ -149,7 +161,7 @@ protected void report() {
     }
 
     @Override
-    public void acquire(long permits) {
+    public synchronized void acquire(long permits) {
```

Comment:
Oh yes, I forgot to make a comment about that but intellij was displaying a warning "Non-atomic operation on volatile field 'permitsAcquired'". Adding `synchronized` seems to fix this issue.

Reply:
I suspect this warning is because of `+=` effectively being two operations on `permitsAcquired`. That warning might possibly be more appropriately addressed by making `permitsAcquired` an `AtomicLong` and using that type's increment methods. That would probably be better than synchronizing on the entire object as this keyword would do.

Reply:
Good idea, I will look into doing that instead.

Reply:
I found an explanation of two reads/writes on the variable at https://stackoverflow.com/a/19744523/196405

Reply:
The most recent commit converts to AtomicLong and its equivalent operations.